modules/markup/markdown/goldmark.go

```diff
@@ -457,12 +462,7 @@ func (r *HTMLRenderer) renderTaskCheckBoxListItem(w util.BufWriter, source []byt
 		} else {
 			_, _ = w.WriteString("<li>")
 		}
-		_, _ = w.WriteString(`<input type="checkbox" disabled=""`)
-		segments := node.FirstChild().Lines()
-		if segments.Len() > 0 {
-			segment := segments.At(0)
-			_, _ = w.WriteString(fmt.Sprintf(` data-source-position="%d"`, segment.Start))
-		}
+		_, _ = w.WriteString(fmt.Sprintf(`<input type="checkbox" disabled="" data-source-position="%d"`, n.SourcePosition))
```

Comment:
`fmt.Fprint(w, ...)`?

Reply:
Seems out of scope for this PR. There are lots of writes using the same method here. 

I think it's best to minimize the changes for this bugfix, since it will need to be backported.